### DISCUSSION TITLE
Circumventing various page caches to get acurate an realtime search results

### DISCUSSION THREAD

**Discussion by SjoerdV** [microsoft-search/pnp-modern-search#1994] [Q&A]
I have noticed that quite a bit of caching is going on when a PnP Search Result webpart is added to a page.

This caching is also applied to the actual execution of the 'query template' sadly, this will cause stale results (or no results) to linger. This cache seems to persist to at least a half hour, which is far to long.

The only way to fix this is entering the browser DevTools F12, go to the Network tab and select 'disable cache' option, and then refresh the page.

My question: can this also be achieved by means of adding a **URL parameter** to the page URL perhaps, so that the page will always get latest query results from the index and not a cached copy? All the other assets and CDN caching is fine, but I really want to see only accurate search results, so I would accept a solution that bypasses ALL the page caches to achieve this.

**Comment by wobba** [comment]
Not sure what caching you refer to, as we don't cache search results in the web part. Every F5 or navigate back and forth to a page should re-run the query. Would be good if you can share some more details of your setup (version, web partc etc.) and how to replicate this.

**Reply by SjoerdV** [reply]
That's great to know! So no caching inside the Result webpart.

Turns out the index is not updated when data/values in additional columns surfaced through lookup columns change. So I had to reindex the connected list manually... 

Now I need to build something that will trigger a re-indexation of a list on regular intervals.

There's always something, right 😉

**Reply by wobba** [reply]
There are no guarantees on the time it takes to index changes from an item unfortunately :( And you should not trigger re-indexing, as that will make indexing worse all-up. If this happens all the time for a list I suggest you open up a support ticket.

**Reply by SjoerdV** [reply]
I think you misunderstand, the issue is actually confirmed here: https://techcommunity.microsoft.com/t5/sharepoint/getting-changes-to-lookups-to-trigger-a-crawl/m-p/634574

So changes to ListA used as a source for Lookup (additional) column in ListB do not trigger a recrawl of ListB, leaving stale results that will only be resolved automatically after the weekly Full crawl schedule (which is not a practical schedule for a real world application).

But it is manageable...

**Reply by wobba** [reply]
That makes sense, and missed the point of about lookup columns.
